src/battle/overlay_12_0224E4FC.c

```diff
@@ -6478,3 +6488,435 @@ BOOL CheckStatusEffectsSubstitute(BATTLECONTEXT *ctx, int battlerId, int status)
+            break;
+        case HOLD_EFFECT_ARCEUS_BUG:
+            type = TYPE_BUG;
+            break;
+        case HOLD_EFFECT_ARCEUS_ROCK:
+            type = TYPE_ROCK;
+            break;
+        case HOLD_EFFECT_ARCEUS_GHOST:
+            type = TYPE_GHOST;
+            break;
+        case HOLD_EFFECT_ARCEUS_DRAGON:
+            type = TYPE_DRAGON;
+            break;
+        case HOLD_EFFECT_ARCEUS_DARK:
+            type = TYPE_DARK;
+            break;
+        case HOLD_EFFECT_ARCEUS_STEEL:
+            type = TYPE_STEEL;
+            break;
+        default:
+            type = TYPE_NORMAL;
+            break;
+        }
+    }
+    
+    return type;
+}
+
+static void ov12_02258584(BATTLECONTEXT *ctx, u8 battlerId) {
+    for (int i = 0; i < LEARNED_MOVES_MAX; i++) {
+        ctx->unk_334.unk1C[battlerId][i] = 0;
+    }
+}
+
+static void ov12_0225859C(BATTLECONTEXT *ctx, u8 battlerId) {
+    ctx->unk_334.unk5C[battlerId] = 0;
+}
+
+static void ov12_022585A8(BATTLECONTEXT *ctx, u8 battlerId) {
+    ctx->unk_334.unk60[battlerId] = 0;
+}
+
+static int ov12_022585B8(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerIdTarget1, int battlerIdTarget2) {
+    int ret = BATTLER_NONE;
+    
+    if (ctx->battleMons[battlerIdTarget1].ability != ABILITY_FORECAST &&
+        ctx->battleMons[battlerIdTarget1].ability != ABILITY_TRACE &&
+        ctx->battleMons[battlerIdTarget1].ability != ABILITY_MULTITYPE &&
+        ctx->battleMons[battlerIdTarget1].hp && ctx->battleMons[battlerIdTarget2].hp &&
+        ctx->battleMons[battlerIdTarget2].ability != ABILITY_FORECAST &&
+        ctx->battleMons[battlerIdTarget2].ability != ABILITY_TRACE &&
+        ctx->battleMons[battlerIdTarget2].ability != ABILITY_MULTITYPE) {
+        if (BattleSystem_Random(bsys) & 1) {
+            ret = battlerIdTarget2;
+        } else {
+            ret = battlerIdTarget1;
+        }
+    } else if (ctx->battleMons[battlerIdTarget1].ability != ABILITY_FORECAST &&
+               ctx->battleMons[battlerIdTarget1].ability != ABILITY_TRACE &&
+               ctx->battleMons[battlerIdTarget1].hp &&
+               ctx->battleMons[battlerIdTarget1].ability != ABILITY_MULTITYPE) {
+        ret = battlerIdTarget1;
+    } else if (ctx->battleMons[battlerIdTarget2].ability != ABILITY_FORECAST &&
+              ctx->battleMons[battlerIdTarget2].ability != ABILITY_TRACE &&
+              ctx->battleMons[battlerIdTarget2].hp &&
+              ctx->battleMons[battlerIdTarget2].ability != ABILITY_MULTITYPE) {
+       ret = battlerIdTarget2;
+   }
+   
+   return ret;
+}
+
+extern u16 ov12_0226CB64[6];
+
+static BOOL ov12_0225865C(BATTLECONTEXT *ctx, int moveNo) {
+    for (int i = 0; i < NELEMS(ov12_0226CB64); i++) {
+        if (ov12_0226CB64[i] == ctx->unk_334.moveData[moveNo].effect) {
+            return TRUE;
+        }
+    }
+    
+    return FALSE;
+}
+
+static int BattleSystem_GetMoveType(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId, int moveNo) {
```

Comment:
this is only for special/type changing moves, might be an idea to specify that

Reply:
aye aye captin